Assets/Mirror/Runtime/LocalConnections.cs

```diff
@@ -44,16 +40,46 @@ public override void Disconnect()
         }
     }
 
+    internal class LocalConnectionBuffer
+    {
+        readonly NetworkWriter writer = new NetworkWriter();
+        readonly NetworkReader reader = new NetworkReader(default(ArraySegment<byte>));
+        int packetCount;
```

Comment:
why do we need packetCount?
what if we just have a bool GetNextPacket(out segment) that tries to read the next bytesandsize segment?
this way we have less complexity (no extra packet count and no haspackets function)

Reply:
I dont think we can know if the next bytes in the array is a array segment or not? 

Without packet count we could read the next 4 bytes and assume it is the length when it could just be left over bytes from last frame's writing

Reply:
Write method always writes bytes+size segment, so if read always reads bytes+size segment then there will be no leftovers.
I suggested this because that's how our transports work too. they do exactly the same, read size+bytes in GetNextMessage bool.
it would be more consistent

Reply:
With the transport they send size of total packet, for example it knows there is only 200 bytes in the packet. But here we are just using a buffer that is atleast 1500 bytes long. So need something to say "there is only X array segments".

Reply:
@James-Frowen can you please add that as a comment above packetCount? it's not obvious.
then we can merge :)